src/middlewared/middlewared/plugins/sysdataset.py

```diff
@@ -366,3 +370,154 @@ def __createlink(self, syspath, item):
                 await self.middleware.call('service.start', i)
 
         await self.__nfsv4link(config)
+
+    @private
+    def remove(self, path, link_only=False):
+        if os.path.exists(path):
+            if os.path.islink(path):
+                os.unlink(path)
+            elif os.path.isdir(path) and not link_only:
+                shutil.rmtree(path)
+            elif not link_only:
+                os.remove(path)
+
+    @private
+    def get_members(self, tar, prefix):
+        for tarinfo in tar.getmembers():
+            if tarinfo.name.startswith(prefix):
+                tarinfo.name = tarinfo.name[len(prefix):]
+                yield tarinfo
+
+    @private
+    def update_collectd_dataset(self):
+        config = self.middleware.call_sync('systemdataset.config')
+        is_freenas = self.middleware.call_sync('system.is_freenas')
+        rrd_mount = ''
+        if config['path']:
+            rrd_mount = f'{config["path"]}/rrd-{config["uuid"]}'
+
+        use_rrd_dataset = False
+        if (
+            rrd_mount and config['rrd'] and (
+                is_freenas or (not is_freenas and self.middleware.call_sync('notifier.failover_status') != 'BACKUP')
+            )
+        ):
+            use_rrd_dataset = True
+
+        # TODO: FreeNAS Config MD5 is hardcoded, this should be looked in future to see if we can change that
+        # If there is a failover table remove the rc.conf cache rc.conf.local will run again using the correct
```

Comment:
What is "failover table"?

"If there is xxx, remove the rc.conf cache; rc.conf.local will run again using the correct collectd_enable" - this can be understood ambiguously without punctuation.